docs/specifications/tests/Zone-TP/zone09.md

```diff
@@ -1,47 +1,276 @@
 
-1. Query the MX record from a delegated name server for the domain.
-2. If the answer from step 1 is not authoritative, iterate step 1 until there is an authoritative answer.
-3. If there are MX records present in the answer, this part of the test
-   case succeeds.
-4. If there are no MX records present in the answer from the MX query,
-   query for the A and AAAA records for the domain.
-5. If there is any A och AAAA record in the answers from the queries in
-   step 4, this test case succeeds.
+It is therefore expected for every domain (zone), excluding those
+excepted above, to publish an MX record in apex of the zone (i.e.
+in the same position as the SOA record). If MX is not present, SMTP
+*can* deliver email using an address record (A or AAAA) as specified
+in [RFC 5321][RFC 5321#section-5.1], section 5.1, but that possibility
+is not in common use. This test case only checks for MX record since
+that is the absolutely dominating way to configure support for email.
+If an address record is found in apex it is more often used for web
+service rather than mail service.
 
-### Outcome(s)
+[RFC 7505] standardizes "Null MX" which in means that there is no
+mail service for the domain. If a "Null MX" is found, it is considered
+to be the same as there is no MX.
 
-If there is no target host (MX, A or AAAA record) to deliver e-mail for the
-domain name, this test case fails.
+In this test case, the follwoing zone types are excepted from the
+requirement of MX:
 
-### Special procedural requirements
+* Root zone
+* TLD zone
+* Zone in the .ARPA tree
 
-None.
+The following zone type is expected not to have an MX (considered harmful):
+
+* TLD zone
+
+
+## Scope
+
+It is assumed that *Child Zone* is tested and reported by [Basic04]. This test
+case will just ignore non-responsive name servers or name servers not giving a
+correct DNS response for an authoritative name server.
+
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+
+## Summary
+
+* Basic name server problems are not reported by this test case.
+* A notify is issues if MX is missing, except for root, an ARPA zone and a TLD.
+* A warning is issued if a TLD *has* an MX.
+
+Message Tag outputted               | Level   | Arguments               | Description of when message tag is outputted
+:-----------------------------------|:--------|:------------------------|:--------------------------------------------
+Z09_INCONSISTENT_MX_RRSETS          | WARNING |                         | Some name servers returns MX RRset and other none.
+Z09_MX_FOUND                        | INFO    | ns_ip_list              | List of name servers that return MX RRset.
+Z09_MX_DATA                         | INFO    | ns_ip_list, domain_list | The mail targets in the MX RRset returned by some name servers, and a list of those.
+Z09_NON-AUTHORITATIVE_MX_RESPONSE   | WARNING | ns_ip_list              | List of name servers that gave non-AA response on MX query.
```

Comment:
"NON-AUTHORITATIVE": shouldn't it be a "_" -> "NON_AUTHORITATIVE" ?

I guess that "non-AA response" stands for "non authoritative response" (am I right?). Is the phrasing "non-AA response" common in the DNS community?